lib/rubocop/cop/mixin/tokens_optimized.rb

```diff
@@ -0,0 +1,17 @@
+module RuboCop
```

Comment:
Right, building a tokens index should make a good difference. I'm curious if we really need to use tokens though. If we do, the best would be to share the cache with a `Force`, even though they haven't really be designed for this.